containers/ecr-viewer/src/app/components/AutoSignout.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+import { useEffect, useRef, useState } from "react";
+
+import {
+  ButtonGroup,
+  ModalFooter,
+  ModalHeading,
+  ModalToggleButton,
+} from "@trussworks/react-uswds";
+import { signOut, useSession } from "next-auth/react";
+
+import Modal, { ModalRef } from "./modal/Modal";
+
+// Events we consider "activity" to keep the session alive
+const ACTIVE_EVENTS = ["click", "keydown", "scroll"];
+
+// Seconds out from session expiration at which we'll warn the user
+const WARNING_DURATION = 90;
+
+const signOutGoHome = () => signOut({ callbackUrl: "/ecr-viewer" });
+/**
+ * Root layout for the view-data page
+ * @returns laid out content
+ */
+export const AutoSignout = () => {
+  const { update, data } = useSession();
+  const [isActive, setIsActive] = useState(false);
+  const [timeToExpireSecs, setTimeToExpireSecs] = useState(99999999999);
+  const modalRef = useRef<ModalRef>(null);
+
+  // flip isActive to true on an event that we mark as activity
+  useEffect(() => {
+    if (isActive) return;
+
+    const updateActive = () => setIsActive(true);
+
+    ACTIVE_EVENTS.forEach((e) => window.addEventListener(e, updateActive));
+    return () =>
+      ACTIVE_EVENTS.forEach((e) => window.removeEventListener(e, updateActive));
+  }, [isActive]);
+
+  // refresh session if becomes active and then don't allow checking activity for 30 seconds
+  useEffect(() => {
+    if (isActive) {
+      // delay session update to make sure original action (e.g. signout) happens first
+      setTimeout(update, 1000);
+      const t = setTimeout(() => {
+        setIsActive(false);
+      }, 30 * 1000);
+
+      return () => clearTimeout(t);
+    }
+  }, [isActive]);
+
+  // Keep track of seconds left until the session expires
+  useEffect(() => {
+    if (data?.expires) {
+      const expires = new Date(data.expires);
+      const ttExpire = Math.floor((expires.valueOf() - Date.now()) / 1000);
+      setTimeToExpireSecs(ttExpire);
+      const t = setTimeout(signOutGoHome, ttExpire * 1000);
+
+      // decrement time each second
+      const i = setInterval(
```

Comment:
Nit: What do you think about using variables names more descriptive than t, i, or even the tt part of ttExpire? My opinion isn't super strong on this, but I tend to like more descriptive names, even if its just "time" "interval" "timeToExpire" etc for readability